core/src/main/java/jenkins/util/ResourceBundleUtil.java

```diff
@@ -72,14 +74,39 @@ private ResourceBundleUtil() {
             return bundleJSON;
         }
 
-        ResourceBundle bundle = ResourceBundle.getBundle(baseName, locale);
+        ResourceBundle bundle = getBundle(baseName, locale, Jenkins.class.getClassLoader());
+        if (bundle == null) {
+            // Not in Jenkins core. Check the plugins.
+            Jenkins jenkins = Jenkins.getInstance();
+            if (jenkins != null) {
+                for (PluginWrapper plugin : jenkins.getPluginManager().getPlugins()) {
+                    bundle = getBundle(baseName, locale, plugin.classLoader);
+                    if (bundle != null) {
+                        break;
+                    }
+                }
+            }
+        }
+        if (bundle == null) {
+            throw new MissingResourceException("Can't find bundle for base name "
+                    + baseName + ", locale " + locale, baseName + "_" + locale, "");
+        }
 
         bundleJSON = toJSONObject(bundle);
         bundles.put(bundleKey, bundleJSON);
 
         return bundleJSON;
     }
 
+    private static ResourceBundle getBundle(String baseName, Locale locale, ClassLoader classLoader) {
```

Comment:
🐜 Add <code>@ChewckForNull</code> to return value and parameters if they are nullable